src/main/java/de/tum/in/www1/artemis/service/iris/IrisSessionService.java

```diff
@@ -113,4 +160,184 @@ else if (session instanceof IrisHestiaSession) {
             throw new BadRequestException("Unknown Iris session type " + session.getClass().getSimpleName());
         }
     }
+
+    /**
+     * Creates the initial system message for the session and set LLM
+     *
+     * @param session The session to generate the initial system message for
+     * @return The created IrisMessage for storing in the database
+     */
+    public IrisMessage createInitialSystemMessage(IrisChatSession session) {
+        var exercise = session.getExercise();
+        // TODO: Error handling in the future
+        String title = "undefined";
+        if (exercise.getCourseViaExerciseGroupOrCourseMember() != null && exercise.getCourseViaExerciseGroupOrCourseMember().getTitle() != null) {
+            title = exercise.getCourseViaExerciseGroupOrCourseMember().getTitle();
+        }
+        String initialSystemMessage = INITIAL_PROMPT_TEMPLATE.formatted(title, ProblemStatementUtils.truncateProblemStatement(exercise.getProblemStatement()));
+
+        var messageContent = new IrisMessageContent();
+        messageContent.setTextContent(initialSystemMessage);
+        var message = new IrisMessage();
+        message.setContent(List.of(messageContent));
+        message.setSender(IrisMessageSender.ARTEMIS);
+        message.setSentAt(ZonedDateTime.now());
+        message.setSession(session);
+        return message;
+    }
+
+    /**
+     * Utility class for removing all unnecessary information from problem statements
+     */
+    private static class ProblemStatementUtils {
```

Comment:
small suggestion: I think this class can be in another file on its own

Reply:
I though about it but I didn't want to put the class with other util methods since the functionality is pretty specific to our use case. If anyone tells me where I can put it or if I should put it in service/util for example I will gladly do it